### PR TITLE
Fix Suggestion Provider for Crossref during autocompletion disable

### LINKED ISSUE #8145: "Crossref" field should also work if Autocomplete is disabled
### JabRef version

Latest development branch build (please note build date below)

### Operating system

Windows

### Details on version and operating system

Windows 11 Pro (build 22471)  |  JabRef 5.4--2021-10-11--2e71d5f

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Enter a citekey (in my case an "_mvBook_") of the parent entry in the **crossref** field
2. The BibLaTeX source doesn't show the crossref line.
3. After closing JabRef and reopening, the **crossref** input field is also empty again.

### Appendix

...
<details>
<summary>Log File</summary>
`It seems nothing is being logged
`</details>

### PR DESCRIPTION
**Pull request by pal-anish**
Closes #8145 

Fixed the initial issue when autocompletion was disabled, the crossref field was not populating the suggestions.
Now the CrossRef field is populating the suggestions even if AutoCompletion(File > Preferences > Autocompletion) is disabled
Auto_Disable
Op1

### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/LibraryTab.java [hunk 1]
```diff
@@ -563,7 +563,8 @@ private void setupAutoCompletion() {
             suggestionProviders = new SuggestionProviders(getDatabase(), Globals.journalAbbreviationRepository, autoCompletePreferences);
         } else {
             // Create empty suggestion providers if auto-completion is deactivated
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/LibraryTab.java:565]: Delete the commented as it's wrong and also the commented out code

File: src/main/java/org/jabref/gui/autocompleter/SuggestionProviders.java [hunk 2]
```diff
@@ -22,14 +22,26 @@ public SuggestionProviders(BibDatabase database, JournalAbbreviationRepository a
...
 
     public SuggestionProvider<?> getForField(Field field) {
         if (isEmpty || !autoCompletePreferences.getCompleteFields().contains(field)) {
+            Set<FieldProperty> fieldProperties = field.getProperties();
+            if (fieldProperties.contains(FieldProperty.SINGLE_ENTRY_LINK)) {
+                return new BibEntrySuggestionProvider(database);
+            }
             return new EmptySuggestionProvider();
         }
+//        if (isEmpty || !autoCompletePreferences.getCompleteFields().contains(field)) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/autocompleter/SuggestionProviders.java:42]: Remove commented out code, that's why we have git, you can always go back in the history and restore deleted code.

**Comment by Siedlerchr** [comment]
Generally looks good, just remove the commented out code

**Comment by pal-anish** [comment]
@Siedlerchr I have made all the changes you suggested please review those. Thanks for your suggestions.

File: src/main/java/org/jabref/gui/autocompleter/SuggestionProviders.java (hunk 2 above)
Comment (pal-anish) [reply src/main/java/org/jabref/gui/autocompleter/SuggestionProviders.java]: Removed those in new commit

File: src/main/java/org/jabref/gui/LibraryTab.java (hunk 1 above)
Comment (pal-anish) [reply src/main/java/org/jabref/gui/LibraryTab.java]: Removed those in new commit

**Review by Siedlerchr — DISMISSED** [review]

**Comment by Siedlerchr** [comment]
Can you please add a changelog entry? (I edit/push my changes to your fork)
And next time please open a new branch

```
- We fixed an issue where the CrossRef field did not work if autocompletion was disabled [#8145](https://github.com/JabRef/jabref/issues/8145)
 ```

**Comment by pal-anish** [comment]
@Siedlerchr I will keep that in mind to open a new branch next time.

**Comment by koppor** [comment]
The code is OK. However, the requirements at https://github.com/JabRef/jabref/issues/8145 are a mess. 

I think, the preferences should be modified to be more user-friendly. Will file an issue in our melting pot.

**Pull request merged**